### PR TITLE
fix(material/form-field): flickering when hovering invalid input on chrome

### LINKED ISSUE #21612: fix(mat-form-field mat-error mat-label): labels and errors on form fields are affecting one another on focus / hover with noticeable opacity applied on transition
#### Reproduction
Use StackBlitz to reproduce your issue: 
- Components: https://stackblitz.com/edit/angular-k6z7ve

Steps to reproduce:
1. Use dark theme in material (this step may not be required, but it's more obvious when viewing it)
2. Click each form field and click out of it to invoke 'required' error on all of them. Then click each individual one and notice the transition styles being applied on other ones.
 
#### Expected Behavior
Clicking each form field should cause a transition and style change that does not affect other form fields.

#### Actual Behavior
Clicking each form field seems to apply styles on other ones on transition when the field is focused or hovered. In particular, it seems like there is some hierarchy in which the styles are applied. 

i.e. Clicking the first form field will apply transition styles on every element. Clicking the second form field will apply styles on itself (not the first field) and every element after, and so on.

#### Environment
- Angular: 11.0.4
- CDK/Material: 11+
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

On a similar topic, is there a possible way to override the styles to not apply these focus/hover styles? I can't find the place where the styles are being applied but it seems to give some opacity and make the font weight smaller. In darker themes, this transition style seems very obvious and looks weird. I'd rather the style be consistent during transition or not apply it at all.

### PR DESCRIPTION
**Pull request by crisbeto**
Fixes some flickering in all adjacent form fields that happens when a specific form field is in its invalid state and the user hovers over it.

Fixes #21612.

### CODE REVIEW

File: src/material/form-field/form-field.scss [hunk 1]
```diff
@@ -184,7 +184,9 @@ $mat-form-field-default-infix-width: 180px !default;
 .mat-form-field.mat-form-field-invalid {
   .mat-form-field-ripple {
     opacity: 1;
-    transform: scaleX(1);
+    // Note that we transition this to `none`, rather than `scaleX(1)`, because the scaling
```
Comment (mmalerba) [inline src/material/form-field/form-field.scss:187]: Have you confirmed that this doesn't mess up any transitions or animations on the element? IIRC things don't animate nicely from `none` to something

Comment (crisbeto) [reply src/material/form-field/form-field.scss:187]: I tried toggling it on/off and it did transition as expected. FWIW, this isn't the only place where we transition to `transform: none`, we do it in other places as a workaround to an issue where text becomes blurry on IE.

Comment (mmalerba) [reply src/material/form-field/form-field.scss:187]: Ok, maybe transform isn't affected by that quirk, I know things like `width` and `height` are.

**Review by mmalerba — APPROVED** [review]
LGTM

**Pull request merged**